### PR TITLE
Add some luamtev examples

### PR DESCRIPTION
**Pull request by HeinrichHartmann**
I am not sure that this repo is the right place for the example script. The reason I am posting this here at this stage is to make adding in-line comments easy using GitHubs review features.

There are a couple of open questions in hello-world.lua I would appreciate feedback to.

### CODE REVIEW

File: examples/luamtev/hello-world.lua [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+-- The main function is executed concurrently by all lua threads
+-- when <concurrent>true</concurrent> is set in the config
+--
+-- Q: How to configure the threaded concurrency?
+-- Apparently <default_queue_threads>10</default_queue_threads> in eventer/config is not doing it.
+--
+function main()
+  -- Get parameters form the environment variables:
+  -- Q: How to pass command line arguments to luamtev? ... is that possible at all?
+  local N = tonumber(os.getenv("LUA_COROS")) or 10
```
Comment (postwait) [inline examples/luamtev/hello-world.lua]: args global variable?

File: examples/luamtev/hello-world.lua [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+-- cache pid/tid
+local pid, tid = mtev.thread_self()
+
+-- initialize RNG
+local s,us = mtev.gettimeofday()
+math.randomseed(us)
+
+function coro_main(coro_id, w)
+  -- Q: Is there any way to get the identity of the currently running coroutine?
+  --    like mtev.thread_self() just for co-routines?
```
Comment (postwait) [inline examples/luamtev/hello-world.lua]: No. You'd use a lua native one?  Does such a thing exist?

File: examples/luamtev/hello-world.lua [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+  mtev.notify("SLEEP", coro_id, slept:seconds())
+
+  -- do some async I/O from the co-routine
+  local proc = mtev.spawn("/bin/echo", { "echo", "Hello from echo" })
+  local status, errno = proc:wait(1)
+  mtev.notify("PROC", coro_id)
+  -- Q: How can we communicate with the subprocess?
+  --    - read from stdout/stderr
+  --    - write to stdin
+
```
Comment (postwait) [inline examples/luamtev/hello-world.lua]: Standard ways yes. In/out/err and kill

File: examples/luamtev/hello-world.lua [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+  -- tell main() we are done
+  mtev.notify("DONE", coro_id)
+end
+
+--
+-- The main function is executed concurrently by all lua threads
+-- when <concurrent>true</concurrent> is set in the config
+--
+-- Q: How to configure the threaded concurrency?
+-- Apparently <default_queue_threads>10</default_queue_threads> in eventer/config is not doing it.
```
Comment (postwait) [inline examples/luamtev/hello-world.lua]: It's is the event loop concurrency not the jobq concurrency.

File: examples/luamtev/hello-world.lua [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+    mtev.log("out", "- found %s\n", x)
+  end
+  -- Discussion: Inter-thread communication seems to be pretty limited.
+  -- For mtev.shared_set/get():
+  -- - only string values are supported
+  -- - no atomic operations "upsert" (mtev.shared_inc?)
+  -- - no locking to avoid races (mtev.shared_lock()/shared_unlock()?)
+  -- Q: Are there any other possibilities to communicate between threads?
+  --    E.g. locks, semaphores, queues?
+
```
Comment (postwait) [inline examples/luamtev/hello-world.lua]: The shared stuff was started on without a spec and abruptly stopped. It is unused and,  I thought,  undocumented. It needs to be designed and built from scratch

File: examples/luamtev/hello-world.lua [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+    mtev.sleep(.1)
+  end
+
+  -- Calling os.exit() terminates all threads.
+  -- If we don't call os.exit() explicilty the process will keep running.
+  -- Q: Is there a better way to "join lua threads and exit" ?
+  --    Maybe:
+  --    -> expose syncrhonization primitives, e.g. semaphores???
+  --    -> Explicit config option to terminate process after main() has returned on all threads?
+  --    -> Ignore the problem (concurrent one-shot applications might not common)
```
Comment (postwait) [inline examples/luamtev/hello-world.lua]: Just like C. Exit is exit, joining threads is the responsibility on the developer.

File: examples/luamtev/hello-world.lua (hunk 1 above)
Comment (HeinrichHartmann) [reply examples/luamtev/hello-world.lua]: right!

File: examples/luamtev/hello-world.lua (hunk 2 above)
Comment (HeinrichHartmann) [reply examples/luamtev/hello-world.lua]: There is `coroutine.running()`. Cool.

File: examples/luamtev/hello-world.lua (hunk 3 above)
Comment (HeinrichHartmann) [reply examples/luamtev/hello-world.lua]: Found examples in mtevbusted/child.lua. Cool.

File: examples/luamtev/hello-world.lua (hunk 4 above)
Comment (HeinrichHartmann) [reply examples/luamtev/hello-world.lua]: Which config stanza would that require?

File: examples/luamtev/hello-world.lua (hunk 5 above)
Comment (HeinrichHartmann) [reply examples/luamtev/hello-world.lua]: OK.

File: examples/luamtev/hello-world.lua (hunk 6 above)
Comment (HeinrichHartmann) [reply examples/luamtev/hello-world.lua]: os.exit() behaves as expected. But how can I do the join from within luamtev?
I don't have access to a parent process where I could call wait() from...
Probably this has to wait until the "shared stuff" landed.

**Comment by HeinrichHartmann** [comment]
Merging with the following changes since review:
- Moved examples to existing src/examples folder
- Rebased to master
- Re-generated documentation

**Pull request merged**